src/submission/exceptions/exceptions.ts

```diff
@@ -308,54 +298,20 @@ export async function getExceptionManifestRecords(
 			index === donorArray.findIndex((donor) => donor.donorId === donorRecord.donorId),
 	);
 
-	// Exceptions only store submitterIds, so all submitterIds have to be collected before we can filter exceptions
-	const submitterDonorIds = donors.map((donor) => donor.submitterId);
+	const { programException, entityException } = await queryForExceptions(programId);
 
-	const { programException, entityException: entityPropertyException } = await queryForExceptions(
-		programId,
-	);
+	const programExceptions = programException?.exceptions || [];
```

Comment:
great error checking! there's also error checking in the `createExceptionManifest` function.
could be good to do all the checking for ` {
		programExceptions,
		entityException,
		missingEntityException,
		treatmentDetailException,
	});` first and then pass only what you need into the `create` function

Reply:
(same kind of error checking, ie. check if you got the exception record, check if result is correct etc)

Reply:
That was the original intent, to separate the two functions as you describe. Revisited and updated!